spec/models/spree/customer_return_spec.rb

```diff
@@ -0,0 +1,128 @@
+              vcr: { cassette_name: 'spree/customer_return_entity_exists' } do
+
+        let(:customer_return) { build(:customer_return) }
+        let(:inventory_unit)  { build(:inventory_unit, variant: variant_1) }
+        let(:return_item)     { build(:return_item, inventory_unit: inventory_unit) }
+        let(:shipwire_id)     { returned_order_shipwire_id }
+        let(:return_items)    { [return_item] }
+
+        before do
+          customer_return.return_items = return_items
+          customer_return.order.update_attribute(:shipwire_id, shipwire_id)
+        end
+
+        it 'add an error message' do
+          customer_return.save!
+          expect(customer_return.errors.messages).to have_key(:shipwire)
+        end
+      end
+    end
+
+    context "when shipwire does accept the return" do
+      context 'when return has 1 return item',
+              vcr: { cassette_name: 'spree/customer_return_create_entity_single_return_item' } do
+
+        let(:customer_return) { build(:customer_return) }
+        let(:inventory_unit)  { build(:inventory_unit, variant: variant_1) }
+        let(:return_item)     { build(:return_item, inventory_unit: inventory_unit) }
+        let(:shipwire_id)     { shipped_order_shipwire_id_1 }
+        let(:return_items)    { [return_item] }
+
+        before do
+          customer_return.return_items = return_items
+          customer_return.order.update_attribute(:shipwire_id, shipwire_id)
+        end
+
+        it 'will post to shipwire' do
+          expect(customer_return).to receive(:process_shipwire_return!).and_call_original
+          customer_return.save!
+        end
+
+        it 'will update the shipwire_id' do
+          customer_return.save!
+          expect(customer_return.shipwire_id).not_to be nil
+        end
+      end
+
+      context 'when return has multiple return items',
+              vcr: { cassette_name: 'spree/customer_return_create_entity_multiple_return_items' } do
+
+        let(:customer_return) { build(:customer_return) }
+        let(:inventory_unit_1) { build(:inventory_unit, variant: variant_1) }
+        let(:inventory_unit_2) { build(:inventory_unit, variant: variant_2, order: inventory_unit_1.order) }
+        let(:return_item_1) { build(:return_item, inventory_unit: inventory_unit_1) }
+        let(:return_item_2) { build(:return_item, inventory_unit: inventory_unit_2) }
+        let(:shipwire_id)   { shipped_order_shipwire_id_2 }
+        let(:return_items)  { [return_item_1, return_item_1, return_item_2, return_item_2] }
+
+        before do
+          customer_return.return_items = return_items
+          customer_return.order.update_attribute(:shipwire_id, shipwire_id)
+        end
+
+        it 'will post to shipwire' do
+          expect(customer_return).to receive(:process_shipwire_return!).and_call_original
+          customer_return.save!
+          expect(customer_return.shipwire_id).not_to be nil
```

Comment:
why in the previous content you used 2 scenarios and here just one? what's the difference?